gui/src/app/state/settings/bitcoind.rs

```diff
@@ -370,23 +359,46 @@ impl Setting for RescanSetting {
                 }
             }
             view::SettingsEditMessage::Confirm => {
-                let date_time = if let Some(date) = NaiveDate::from_ymd_opt(
+                let t = if let Some(date) = NaiveDate::from_ymd_opt(
                     i32::from_str(&self.year.value).unwrap_or(1),
                     u32::from_str(&self.month.value).unwrap_or(1),
                     u32::from_str(&self.day.value).unwrap_or(1),
-                ) {
-                    if date < NaiveDate::from_str("2009-01-03").unwrap() {
-                        self.invalid_date = true;
-                        return Command::none();
-                    } else {
-                        self.invalid_date = false;
-                        date
+                )
+                .and_then(|d| d.and_hms_opt(0, 0, 0))
+                .map(|d| d.timestamp() as u32)
```

Comment:
I noticed from the logs that if I put dates earlier than 1970, a future timestamp is passed to the rescan. This shouldn't matter as future timestamps return an error, but perhaps if the timestamp is negative, it should be set to 0 before converting to `u32`?

Reply:
ah good catch !